src/main/java/org/uwdigi/rag/service/AssistantService.java

```diff
@@ -81,14 +88,16 @@ public String getResponse() {
    */
   public QueryResponse processQuery(String query, String modelName) {
     log.debug("Processing with model: {}", modelName);
-
-    ChatLanguageModel chatLanguageModel = this.modelFactory.createModel(modelName);
+    
+    ChatLanguageModel answerModel = modelConfig.getAnswerGenerationModel();
 
     ContentRetriever contentRetriever =
         SqlDatabaseContentRetriever.builder()
             .dataSource(dataSource)
-            .chatLanguageModel(chatLanguageModel)
-            .ollamaChatModel(ollamaChatModel)
```

Comment:
Let's repurpose this parameter to take in the configured one for answer generation.